### DISCUSSION TITLE
Updated geoip documentation 2.4

### DISCUSSION THREAD

**Discussion by Carlos-mb** [Security-Onion-Solutions/securityonion#13451] [2.4] [answered]
### Version

2.4.90

### Installation Method

Security Onion ISO image

### Description

other (please provide detail below)

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

6

### RAM

64

### Storage for /

300

### Storage for /nsm

300

### Network Traffic Collection

span port

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Hi

According to documentation, the way to enable GeoIp is this:

Elasticsearch 8 no longer includes GeoIP databases by default. We include GeoIP databases for Elasticsearch so that all users will have GeoIP functionality. If your search nodes have Internet access and can reach geoip.elastic.co and storage.googleapis.com, then you can opt-in to database updates if you want more recent information. To do this, add the following to your Elasticsearch [Salt](https://docs.securityonion.net/en/2.4/salt.html#salt) config:

config:
ingest:
geoip:
downloader:
enabled: true

But I'm not able to find how to do this change.

I'd like any help because I have found other posts talking about this, but they are related to older SO versions and the files and paths are not present in 2.4

Thanks.

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by cm-ops** [comment]
In SOC, `Administration > Configuration > elasticsearch > advanced` place the below in the config box and hit the green check mark.

```
elasticsearch:
  config:
    ingest:
      geoip:
        downloader:
          enabled: true
```

**Reply by Carlos-mb** [reply]
Thank you.

If I use your suggested text, it creates a new entry in configuration: Aministration -> Configuration -> config -> ingest -> geoip -> downloader -> enabled = true. And I can't change it because it does not accept any value, neither "false" not empty.

I have had to delete the entry from  /opt/so/saltstack/local/pillar/global/soc_global.sls

I have use the text as appears in documentation:

config:
  ingest:
    geoip:
      downloader:
        enabled: true



it doens't matter what I do, I left empty elasticsearch > advanced, I restart the VM, I put it false and then true.... 

I don't see any geographical info in the events and I still see the tag "_geoip_expired_database" in them. 

And I get these errors:

[2024-08-09T17:14:25,933][INFO ][org.elasticsearch.plugins.PluginsService] loaded module [ingest-geoip]
[2024-08-09T17:14:35,474][WARN ][org.elasticsearch.ingest.geoip.GeoIpDownloader] could not delete old chunks for geoip database [GeoLite2-ASN.mmdb]
[2024-08-09T17:14:35,357][ERROR][org.elasticsearch.ingest.geoip.GeoIpDownloader] exception during geoip databases update
[2024-08-09T17:14:35,732][WARN ][org.elasticsearch.ingest.geoip.GeoIpDownloader] could not delete old chunks for geoip database [GeoLite2-Country.mmdb]
[2024-08-09T17:14:35,838][WARN ][org.elasticsearch.ingest.geoip.GeoIpDownloader] could not delete old chunks for geoip database [GeoLite2-City.mmdb]

Thank you so much for your support.

**Comment by Carlos-mb** [comment]
@cm-ops THANKS!

After hours looking for info. Many people said that the trick was to disable and re-enable geoip. 

Finally, I tried this:

1- in Administration > Configuration > elasticsearch > advanced
I wrote FALSE instead of TRUE
```
elasticsearch:
  config:
    ingest:
      geoip:
        downloader:
          enabled: false
```

2- restarted so-elasticsearch-restart
3- Changed false for true
4- Restarted so-elasticsearch-restart

And now It's working fine

I thank your tip

Regards
Carlos

**Comment by Carlos-mb** [comment]
1- in Administration > Configuration > elasticsearch > advanced
I wrote FALSE instead of TRUE

```
elasticsearch:
  config:
    ingest:
      geoip:
        downloader:
          enabled: false
```
2- restarted so-elasticsearch-restart
3- Changed false for true
4- Restarted so-elasticsearch-restart

**Comment by dougburks** [comment, marked as answer]
I've updated the development version of the documentation and it will be included in the next release:
https://docs.securityonion.net/en/dev/elasticsearch.html#geoip

**Reply by Carlos-mb** [reply]
THANKS!

I am honored that you have taken my comment into account.

Regards,
Carlos.
